jupyter_client/channels.py

```diff
@@ -114,30 +114,18 @@ def _poll(self, start_time: float) -> t.List[t.Any]:
         will be an empty list if no messages arrived before the timeout,
         or the event tuple if there is a message to receive.
         """
+        # Check if an event is waiting
+        events = self.poller.poll(0)
+        if events:
+            return events
```

Comment:
I don't think we should check if we have an event at the beginning, we should only wait until_dead and then check.